CyvoreOS/phone_numbers_extension.py

```diff
@@ -0,0 +1,270 @@
+    "+380",  # Ukraine: 44 123 4567 -> +380 44 123 4567
+    "+44",  # United Kingdom: (20) 7946 0958 -> +44 20 7946 0958
+    "+39",  # Vatican City: 06 6989 -> +39 06 6989
+]
+
+
+def find_phone_numbers(data: str) -> List[str]:
+    """
+    Finds Phone Numbers If Exist In Data
+    """
+    rgx_possible = set()
+    logging.info("Looking For Phone Numbers")
+    matches_us = re.findall(USPHONEREG, data)
+    matches_israel = re.findall(ISPHONEREG, data)
+    matches_europe = re.findall(EUPHONEREG, data)
+    logging.debug(
+        "Regex matches - US: %d, Israel: %d, Europe: %d",
+        len(matches_us),
+        len(matches_israel),
+        len(matches_europe),
+    )
+    for match in matches_us + matches_israel + matches_europe:
+        rgx_possible.add((match.strip()).lstrip("+"))
+    return list(rgx_possible)
+
+
+def normalize_phone_number(phone: str) -> str:
+    """
+    Normalizes a phone number by removing non-numeric symbols (except '+')
+    and ensuring it starts with a '+' if not already present.
+            (phonenumbers lib checks numbers better with + at start)
+    """
+    # Remove all characters except digits and '+'
+    normalized = re.sub(r"[^\d\+]", "", phone)
+    logging.debug("After removing non-numeric symbols: %s", normalized)
+
+    # Ensure the phone number starts with '+'
+    if not normalized.startswith("+"):
+        normalized = "+" + normalized
+        logging.debug("Added '+' to the start of the number: %s", normalized)
+
+    logging.info("Final normalized phone number: %s", normalized)
+    return normalized
+
+
+def normalize_phone_numbers(phone_list: PhoneNumbersList) -> PhoneNumbersList:
+    """
+    Normalizes a list of phone numbers, ensuring each number starts with '+'
+    and contains only valid characters for phone numbers.
+    """
+    logging.info(
+        "Starting normalization of phone numbers. Total numbers: %d", len(phone_list)
+    )
+    normal = set()
+    for phone in phone_list:
+        logging.debug("Normalizing number: %s", phone)
+        normalized_phone = normalize_phone_number(phone)
+        normal.add(normalized_phone)
+        logging.debug("Normalized number added to the set: %s", normalized_phone)
+
+    normalized_list = list(normal)
+    logging.info(
+        "Normalization complete. Total unique normalized numbers: %d",
+        len(normalized_list),
+    )
+    return normalized_list
+
+
+# ------------------------------------------------------------------------------------------------------
+def validate_number(num: phonenumbers.PhoneNumber) -> str:
+    """
+    Validates a PhoneNumber object from the phonenumbers library.
+    Checks if the number is possible and valid. Returns the formatted number in E.164 format if valid.
+    """
+    # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    try:
+        logging.debug(
+            "Validating phone number: %s",
+            phonenumbers.format_number(num, phonenumbers.PhoneNumberFormat.E164),
```

Comment:
Function shouldn't be inside log message, separate it into two different lines, and make sure that it really saves the results in num. 
Also you should create a new var for working inside the function and not changing the input var, because I don't know the type of phonenumbers.PhoneNumber but if its some kind of pointer, then changing it could create undesired effects on the code